be/src/storage/tablet_updates.cpp

```diff
@@ -1804,11 +1806,231 @@ Status TabletUpdates::load_from_base_tablet(int64_t request_version, Tablet* bas
+
+    vectorized::Schema base_schema = vectorized::ChunkHelper::convert_schema_to_format_v2(base_tablet->tablet_schema());
+
+    OlapReaderStatistics stats;
+
+    for (int i = 0; i < src_rowsets.size(); i++) {
+        const auto& src_rowset = src_rowsets[i];
+        LOG(INFO) << "begin to convert a history rowset. version=" << src_rowset->version();
+
+        RowsetReleaseGuard guard(src_rowset->shared_from_this());
+        auto beta_rowset = down_cast<BetaRowset*>(src_rowset.get());
+        auto res = beta_rowset->get_segment_iterators2(base_schema, base_tablet->data_dir()->get_meta(),
+                                                       version.major(), &stats);
+        if (!res.ok()) {
+            return res.status();
+        }
+
+        RowsetId rid = StorageEngine::instance()->next_rowset_id();
+
+        RowsetWriterContext writer_context(kDataFormatUnknown, config::storage_format_version);
+        writer_context.rowset_id = rid;
+        writer_context.tablet_uid = _tablet.tablet_uid();
+        writer_context.tablet_id = _tablet.tablet_id();
+        writer_context.partition_id = _tablet.partition_id();
+        writer_context.tablet_schema_hash = _tablet.schema_hash();
+        writer_context.rowset_type = _tablet.tablet_meta()->preferred_rowset_type();
+        writer_context.rowset_path_prefix = _tablet.schema_hash_path();
+        writer_context.tablet_schema = &_tablet.tablet_schema();
+        writer_context.rowset_state = VISIBLE;
+        writer_context.version = src_rowset->version();
+        writer_context.version_hash = src_rowset->version_hash();
+        writer_context.segments_overlap = src_rowset->rowset_meta()->segments_overlap();
+
+        std::unique_ptr<RowsetWriter> rowset_writer;
+        status = RowsetFactory::create_rowset_writer(writer_context, &rowset_writer);
+        if (!status.ok()) {
+            LOG(INFO) << "build rowset writer failed";
+            return Status::InternalError("build rowset writer failed");
+        }
+
+        status = _convert_from_base_rowset(base_tablet, res.value(), chunk_changer, rowset_writer);
+        if (!status.ok()) {
+            LOG(WARNING) << "failed to convert from base rowset, exit alter process";
+            return status;
+        }
+
+        _tablet.obtain_push_lock();
+        std::shared_ptr<Rowset> new_rowset = rowset_writer->build();
+        if (new_rowset == nullptr) {
+            LOG(WARNING) << "failed to build rowset, exit alter process";
+            _tablet.release_push_lock();
+            return Status::InternalError("failed to build rowset, exit alter process");
+        }
+
+        _tablet.release_push_lock();
+
+        auto& new_rowset_load_info = new_rowset_load_infos[i];
+        // use src_rowset's meta as base, change some fields to new tablet
+        auto& rowset_meta_pb = new_rowset_load_info.rowset_meta_pb;
+        src_rowset->rowset_meta()->to_rowset_pb(&rowset_meta_pb);
```

Comment:
convert_from is different from link_from, the meta change more fields(size, row_count, etc.), should check all fields if they are different, all just build meta from scratch.